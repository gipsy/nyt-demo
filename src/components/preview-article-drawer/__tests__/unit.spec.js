import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import getElementWithContext from 'react-test-context-provider'
import getStore from '@bundles'
import { initialData } from '@tests/initial-data'
// import { store } from '@tests/store'

import PreviewArticleDrawer from '@components/preview-article-drawer'

const activeDrawerContext = {...initialData, news: { ...initialData.news, drawer: true }}
const store = getStore(activeDrawerContext)
describe('PreviewArticleDrawer', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element
    })
  })

  afterEach(() => {
    ReactDOM.createPortal.mockClear()
  })

  test('renders without crashing', () => {
    const el = getElementWithContext({ store }, <PreviewArticleDrawer />)
    const component = renderer.create(el).toJSON()
    console.log(component)
    expect(component).toMatchSnapshot()
  })

  // it('renders actual component', () => {
  //   expect(wrap.find(PreviewArticleDrawer).exists()).toBeTruthy()
  // })
})
