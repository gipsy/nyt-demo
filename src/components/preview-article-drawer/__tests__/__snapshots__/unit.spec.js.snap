// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PreviewArticleDrawer renders without crashing 1`] = `
.c0 {
  font-family: 'Noto Serif KR',sans-serif;
  font-size: 28px;
  color: #000;
}

.c3 {
  display: inline-block;
  background-color: #ececec;
  width: 100%;
}

.c4 {
  font-size: 16px;
  color: #000;
  margin-top: 20px;
}

.c1 {
  text-transform: uppercase;
  font-size: 12px;
  color: #ddd;
  display: inline-block;
}

.c2 {
  font-family: 'Helvetica',sans-serif;
  font-weight: 500;
  font-size: 14px;
  display: inline-block;
  float: right;
}

@media (max-width:600px) {
  .c0 {
    font-size: 23px;
  }
}

<div
  className="ant-drawer ant-drawer-right ant-drawer-open"
  onTransitionEnd={[Function]}
  style={
    Object {
      "zIndex": undefined,
    }
  }
>
  <div
    className="ant-drawer-mask"
    onClick={[Function]}
    style={Object {}}
  />
  <div
    className="ant-drawer-content-wrapper"
    style={
      Object {
        "height": undefined,
        "msTransform": "",
        "transform": "",
        "width": "50%",
      }
    }
  >
    <div
      className="ant-drawer-content"
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="ant-drawer-wrapper-body"
        onTransitionEnd={[Function]}
        style={
          Object {
            "height": "100%",
            "overflow": "auto",
          }
        }
      >
        <button
          aria-label="Close"
          className="ant-drawer-close"
          onClick={[Function]}
        >
          <span
            className="ant-drawer-close-x"
          >
            <i
              className="anticon anticon-close"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="close"
                fill="currentColor"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                />
              </svg>
            </i>
          </span>
        </button>
        <div
          className="ant-drawer-body"
        >
          <h2
            className="c0"
          >
            Singapore, for Those Who Aren’t Crazy Rich
          </h2>
          <div
            className="ant-divider ant-divider-horizontal"
          />
          <span
            className="c1"
          >
            The New York Times
          </span>
          <time
            className="c2"
          >
            Sep 21, 2018
          </time>
          <img
            className="c3"
            src="https://graphics8.nytimes.com/images/2018/09/23/travel/23singapore1/merlin_142944708_9276e9dd-b36e-40f3-90f6-34e79e64552e-jumbo.jpg"
          />
          <div
            className="ant-row"
          >
            <div
              className="ant-col-24"
            >
              <p
                className="c4"
              >
                In the hit movie “Crazy Rich Asians,” the country looks luxe. But it’s easy to have a great time (and eat memorably well) without spending much.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
